Assets/Scripts/Contracts/PepemonCardDeck.cs

```diff
@@ -119,13 +137,15 @@ public static async Task<string> RemoveBattleCard(ulong deckId)
 
     /// <summary>
     /// Approval is necessary to prevent this error in some cases: ERC1155#safeTransferFrom: INVALID_OPERATOR
+    /// Note: The contract that needs this is PepemonFactory, not PepemonCardDeck
     /// </summary>
     /// <param name="approved">Approval state to allow moving cards</param>
     /// <returns>Transaction result</returns>
     public static async Task<string> SetApprovalState(bool approved)
     {
         var approvalRequest = Web3Controller.instance.GetContractTransactionUnityRequest();
         return await approvalRequest.SignAndSendTransactionAsync(
+            // Using this ABI is intentional, see the note above
             new Contracts.PepemonFactory.abi.ContractDefinition.SetApprovalForAllFunction()
             {
                 Operator = Address,
```

Comment:
`GetApprovalState` and `SetApprovalState` would be for transferring this NFT (the deck itself is also an NFT), so it would make sense to put these (Set and Get) onto the PepemonFactory instead.

Also, approval is based on the calling contract (the deck in this case), so it would be better to expose the operator argument, e.g. 
`pepemonFactory.SetApprovalForAll(operator = pepemonCardDeck)`
That will be easier to understand for people who are familiar with the contract interfaces